src/sentry/snuba/outcomes.py

```diff
@@ -227,19 +230,22 @@ def __init__(
         if len(raw_fields) == 0:
             raise InvalidField('At least one "field" is required.')
         self.fields = {}
-        self.aggregations = []
         self.query: List[Any] = []  # not used but needed for compat with sessions logic
         start, end, rollup = get_constrained_date_range(query, allow_minute_resolution)
-        self.dataset = _outcomes_dataset(rollup)
+        self.dataset, self.match = _outcomes_dataset(rollup)
         self.rollup = rollup
         self.start = start
         self.end = end
-
+        self.conditions = [
```

Comment:
setting these two items in the conditions list then extending it below feels weird to me.  how about moving these into the get_conditions function? and instead of passing `start` and `end` as parameters, you can move get_conditions to be a method on the QueryDefinition class so it will have access to `start` and `end`